test_pageserver_restarts_under_worload is flaky

```
2023-03-15T11:05:41.4093101Z ==================================== ERRORS ====================================
2023-03-15T11:05:41.4094222Z _________ ERROR at teardown of test_pageserver_restarts_under_worload __________
2023-03-15T11:05:41.4094977Z [gw14] linux -- Python 3.9.2 /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/bin/python
2023-03-15T11:05:41.4095822Z /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/allure_commons/_allure.py:200: in __call__
2023-03-15T11:05:41.4096290Z     return self._fixture_function(*args, **kwargs)
2023-03-15T11:05:41.4096590Z test_runner/fixtures/neon_fixtures.py:1056: in _shared_simple_env
2023-03-15T11:05:41.4096936Z     yield env
2023-03-15T11:05:41.4097181Z test_runner/fixtures/neon_fixtures.py:853: in __exit__
2023-03-15T11:05:41.4097475Z     self.env.pageserver.assert_no_errors()
2023-03-15T11:05:41.4098351Z test_runner/fixtures/neon_fixtures.py:2172: in assert_no_errors
2023-03-15T11:05:41.4098664Z     assert not errors
2023-03-15T11:05:41.4099250Z E   AssertionError: assert not ['2023-03-15T11:02:36.952113Z ERROR gc_loop{tenant_id=9ad7798ee55e644ea65b8a7d9a353175}: Gc failed, retrying in 2s: Cannot run GC iteration on inactive tenant\n']
```

https://neon-github-public-dev.s3.amazonaws.com/reports/pr-3821/release/4425498523/index.html#suites/542b1248464b42cc5a4560f408115965/a5ad46f5e6569269/

--- a/test_runner/regress/test_pageserver_restarts_under_workload.py
+++ b/test_runner/regress/test_pageserver_restarts_under_workload.py
@@ -20,7 +20,7 @@ def test_pageserver_restarts_under_worload(neon_simple_env: NeonEnv, pg_bin: PgB
     # the background task may complete the init task delay after finding an
     # active tenant, but shutdown starts right before Tenant::gc_iteration
     env.pageserver.allowed_errors.append(
-        r"Gc failed, retrying in \S+: Cannot run GC iteration on inactive tenant"
+        r".*Gc failed, retrying in \S+: Cannot run GC iteration on inactive tenant"
     )
 
     def run_pgbench(pg: Postgres):

--- a/test_runner/regress/test_pageserver_restarts_under_workload.py
+++ b/test_runner/regress/test_pageserver_restarts_under_workload.py
@@ -17,6 +17,12 @@ def test_pageserver_restarts_under_worload(neon_simple_env: NeonEnv, pg_bin: PgB
     n_restarts = 10
     scale = 10
 
+    # the background task may complete the init task delay after finding an
+    # active tenant, but shutdown starts right before Tenant::gc_iteration
+    env.pageserver.allowed_errors.append(
+        r"Gc failed, retrying in \S+: Cannot run GC iteration on inactive tenant"
+    )
+
     def run_pgbench(pg: Postgres):
         connstr = pg.connstr()
         log.info(f"Start a pgbench workload on pg {connstr}")

